Release announcement for 1.9.1 (#9085)
istio/istio.io: content/en/news/security/istio-security-2021-001/index.md

---
title: ISTIO-SECURITY-2021-001
subtitle: Security Bulletin
description: JWT authentication can be bypassed when AuthorizationPolicy is misused.
cves: [N/A]
cvss: "N/A"
vector: ""
releases: ["1.9.0"]
publishdate: 2021-03-1
keywords: [CVE]
skip_seealso: true
---

{{< security_bulletin >}}

> This issue only affects Istio 1.9.0. Previous versions of Istio are not affected. This is rated an 8.2 CVE by Istio.

Reviewer: ```suggestion
This issue only affects Istio 1.9.0. Previous versions of Istio are not affected. This is rated an 8.2 CVSS CVE by Istio.
```

> This issue only affects Istio 1.9.0. Previous versions of Istio are not affected. This is rated an 8.2 CVE by Istio.

Reviewer 2: ```suggestion
This issue only affects Istio 1.9.0; previous versions of Istio are not affected. This issue is rated an 8.2 CVE by the Istio product security working group.
```

Author: Fixed manually

Envoy, and subsequently Istio, is vulnerable to a newly discovered vulnerability:

- [Envoy JWT filter bypass when using the allow_missing configuration under requires_any](https://groups.google.com/g/envoy-security-announce/c/aqtBt5VUor0).

You are subject to the vulnerability if you are using `RequestAuthentication` alone for JWT validation.

You are not subject to the issue if you use both `RequestAuthentication` and `AuthorizationPolicy` for JWT validation.

{{< warning >}}
Please note that `RequestAuthentication` is used to define a list of issuers that should be accepted. It does not reject
a request without JWT token.
{{< /warning >}}

For Istio, this vulnerability only exists if your service:
* Accepts JWT tokens (with `RequestAuthentication`)

> * Some service paths don’t have `AuthorizationPolicy` applied.

Reviewer 2: ```suggestion
* Has some service paths without `AuthorizationPolicy` applied.
```

---
title: ISTIO-SECURITY-2021-001
subtitle: Security Bulletin
description: JWT authentication can be bypassed when AuthorizationPolicy is misused.
cves: [N/A]

> cvss: "N/A"

Reviewer 2: 8.2 ?

Author: Added

vector: ""
releases: ["1.9.0"]

> publishdate: 2021-03-1

Reviewer 2: ```suggestion
publishdate: 2021-03-01
```

Reviewer 2: Hopefully this will fix the fact that it's not sorting correctly in the page list.

keywords: [CVE]
skip_seealso: true
---

{{< security_bulletin >}}

This issue only affects Istio 1.9.0; previous versions of Istio are not affected. This issue has been given a CVSS score
of 8.2 by the Istio product security working group.

Envoy, and subsequently Istio, is vulnerable to a newly discovered vulnerability:

> - [Envoy JWT filter bypass when using the allow_missing configuration under requires_any](https://groups.google.com/g/envoy-security-announce/c/aqtBt5VUor0).

Reviewer 3: ```suggestion
- [Envoy JWT filter bypass when using the `allow_missing` configuration under `requires_any`](https://groups.google.com/g/envoy-security-announce/c/aqtBt5VUor0).
```

You are subject to the vulnerability if you are using `RequestAuthentication` alone for JWT validation.

> You are not subject to the issue if you use both `RequestAuthentication` and `AuthorizationPolicy` for JWT validation.

Reviewer 2: ```suggestion
You are **not** subject to the vulnerability if you use **both** `RequestAuthentication` and `AuthorizationPolicy` for JWT validation.
```

{{< warning >}}
Please note that `RequestAuthentication` is used to define a list of issuers that should be accepted. It does not reject
a request without JWT token.
{{< /warning >}}

For Istio, this vulnerability only exists if your service:
* Accepts JWT tokens (with `RequestAuthentication`)
* Has some service paths without `AuthorizationPolicy` applied.

> If both conditions are met, then an incoming request with a JWT token, and the token issuer is not in

Reviewer 4: ```suggestion
For the service paths that both conditions are met, an incoming request with a JWT token, and the token issuer is not in
```
